pvlib/albedo.py

```diff
@@ -0,0 +1,120 @@
+        Water wave roughness coefficient. [-]
+
+    Returns
+    -------
+    albedo : numeric
+        Albedo for inland water bodies. [-]
+
+    Notes
+    -----
+    The equation for calculating the albedo :math:`\rho` is given by
+
+    .. math::
+       :label: albedo
+
+        \rho = c^{(r \cdot \sin(\alpha) + 1)}
+
+    Inputs to the model are the water color coefficient :math:`c` [-], the
+    water wave roughness coefficient :math:`r` [-] and the solar elevation
+    :math:`\alpha` [degrees]. Parameters are provided in [1]_ , and are coded
+    for convenience in :data:`~pvlib.albedo.WATER_COLOR_COEFFS` and
+    :data:`~pvlib.albedo.WATER_ROUGHNESS_COEFFS`. The values of these
+    coefficients are experimentally determined.
+
+    +------------------------+-------------------+-------------------------+
+    | Surface and condition  | Color coefficient | Wave roughness          |
+    |                        | (:math:`c`)       | coefficient (:math:`r`) |
+    +========================+===================+=========================+
+    | Clear water, no waves  | 0.13              | 0.29                    |
+    +------------------------+-------------------+-------------------------+
+    | Clear water, ripples   | 0.16              | 0.70                    |
+    | up to 2.5 cm           |                   |                         |
+    +------------------------+-------------------+-------------------------+
+    | Clear water, ripples   | 0.23              | 1.25                    |
+    | larger than 2.5 cm     |                   |                         |
+    | (occasional whitecaps) |                   |                         |
+    +------------------------+-------------------+-------------------------+
+    | Clear water,           | 0.30              | 2.00                    |
+    | frequent whitecaps     |                   |                         |
+    +------------------------+-------------------+-------------------------+
+    | Green water, ripples   | 0.22              | 0.70                    |
+    | up to 2.5cm            |                   |                         |
+    +------------------------+-------------------+-------------------------+
+    | Muddy water, no waves  | 0.19              | 0.29                    |
+    +------------------------+-------------------+-------------------------+
+
+    References
+    ----------
+    .. [1] Dvoracek M.J., Hannabas B. (1990). "Prediction of albedo for use in
+       evapotranspiration and irrigation scheduling." IN: Visions of the Future
+       American Society of Agricultural Engineers 04-90: 692-699.
+    """
+
+    if surface_condition is None:
+        if (color_coeff is None) or (wave_roughness_coeff is None):
+            raise ValueError('Either a `surface_condition` has to be chosen or'
+                             ' a combination of `color_coeff` and'
+                             '`wave_roughness_coeff`.')
+    else:
```

Comment:
It may be and edge-case, but what should happen if all the optional parameters are provided? I prefer an exception (or a warning) instead of any parameter being ignored.
Bonus case: what should happen if all optional parameters are not passed a value?

I would change the if blocks to something along the following snippet, IDK what the other reviewers think about it @kandersolar @AdamRJensen 

```python
    if surface_condition is not None and (
        color_coeff is None and wave_roughness_coeff is None
    ):
        # use surface_condition
        pass
    elif surface_condition is None and not (
        color_coeff is None or wave_roughness_coeff is None
    ):
        # use color_coeff and wave_roughness_coeff
        pass
    else:
        raise ValueError(
            "Either a `surface_condition` has to be chosen or"
            " a combination of `color_coeff` and"
            "`wave_roughness_coeff`."
        )
```

PD: sorry for this suggestion at the current progress of the PR

Reply:
@echedey-ls looks sleek to me!

Reply:
The way it is written now if you only give the solar_elevation and omit the rest, you get the ValueError